model.go

```diff
@@ -26,11 +39,146 @@ func LoadModelFile(path string) (*Model, error) {
 	return LoadModel(fp)
 }
 
-func LoadModel(r io.Reader) (*Model, error) {
+func LoadModel(r io.Reader) (Model, error) {
 	dec := json.NewDecoder(r)
 	var m Model
 	if err := dec.Decode(&m); err != nil {
 		return nil, err
 	}
-	return &m, nil
+	for i := range m {
+		m[i].WeightVec = flattenWeight(m[i].Weight)
+	}
+	return m, nil
+}
+
+func (m Model) newPaddedPixels(p *Pixels) *Pixels {
+	px := len(m)
+	ex := NewPixels(p.Width+2*px, p.Height+2*px)
+	for h := 0; h < ex.Height; h++ {
+		for w := 0; w < ex.Width; w++ {
+			i := w + h*ex.Width
+			if w < px { // Left outer area
+				if h < px {
+					ex.Pix[i] = p.Pix[0] // Left upper area
+				} else if px+p.Height <= h {
+					ex.Pix[i] = p.Pix[(p.Height-1)*p.Width] // Left lower area
+				} else {
+					ex.Pix[i] = p.Pix[(h-px)*p.Width] // Left outer area
+
+				}
+			} else if px+p.Width <= w { // Right outer area
+				if h < px {
+					ex.Pix[i] = p.Pix[p.Width-1] // Right upper area
+				} else if px+p.Height <= h {
+					ex.Pix[i] = p.Pix[p.Width-1+(p.Height-1)*p.Width] // Right lower area
+				} else {
+					ex.Pix[i] = p.Pix[p.Width-1+(h-px)*p.Width] // Right outer area
+				}
+			} else if h < px {
+				ex.Pix[i] = p.Pix[w-px] // Upper outer area
+			} else if px+p.Height <= h {
+				ex.Pix[i] = p.Pix[w-px+(p.Height-1)*p.Width] // Lower outer area
+			} else {
+				ex.Pix[i] = p.Pix[w-px+(h-px)*p.Width] // Inner area
+			}
+		}
+	}
+	return ex
+}
+
+func (m Model) Encode(r, g, b *Pixels) (R, G, B *Pixels) {
```

Comment:
__[gometalinter]__
* [golint] exported method Model.Encode should have comment or be unexported